### PR TITLE
Fixes #291

### LINKED ISSUE #291: Unclear error when license attribute is missing from NetCDF file
`license` is a required attribute in MML. If the attribute is not present in the input NetCDF file, the following error message is returned, and nothing else:

```
NetCDF: Attribute not found
``` 

This is not helpful. The user should be notified that the license attribute is missing.

### PR DESCRIPTION
**Pull request by lhmarsden**
**Summary**: I have updated nc_to_mmd.py so that a specific error is returned if there is no license attribute in the input NetCDF file.

**Related issue**: #291 

**Suggested reviewer(s)**: @mortenwh 

**Reviewer checklist**:

- [ ] The headers of all files contain a reference to the repository license (i.e., "License: This file is part of py-mmd-tools, licensed under the Apache License 2.0 (https://www.apache.org/licenses/LICENSE-2.0)")
- [ ] 100% test coverage of new code - meaning:
    - [ ] The overall test coverage increased or remained the same as before
    - [ ] Every function is accompanied with a test suite
    - [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
    - [ ] Functions with optional arguments have separate tests for all options
- [ ] Examples are supported by doctests
- [ ] All tests are passing
- [ ] All names (e.g., files, classes, functions, variables) are explicit
- [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: py_mmd_tools/nc_to_mmd.py [hunk 1]
```diff
@@ -1581,8 +1586,9 @@ def to_mmd(self, collection=None, checksum_calculation=False, mmd_yaml=None,
         mmd_yaml.pop('geographic_extent')
 
         # Get use_constraint data
+        print('BEFORE')
         self.metadata['use_constraint'] = self.get_license(mmd_yaml.pop('use_constraint'), ncin)
-
+        print('AFTER')
```
Comment (mortenwh) [inline py_mmd_tools/nc_to_mmd.py:1591]:
```suggestion
```

File: py_mmd_tools/nc_to_mmd.py [hunk 2]
```diff
@@ -1581,8 +1586,9 @@ def to_mmd(self, collection=None, checksum_calculation=False, mmd_yaml=None,
         mmd_yaml.pop('geographic_extent')
 
         # Get use_constraint data
+        print('BEFORE')
```
Comment (mortenwh) [inline py_mmd_tools/nc_to_mmd.py:1589]:
```suggestion
```

**Review by mortenwh — CHANGES_REQUESTED** [review]
Thanks! One thing: please don't use print statements in the code, but make a test that verifies that the correct error message is provided. I didn't look at the changes in detail but will do that once you've added a test.

File: py_mmd_tools/nc_to_mmd.py (hunk 2 above)
Comment (lhmarsden) [reply py_mmd_tools/nc_to_mmd.py]: Removed - sorry, I forgot to take this out.

File: py_mmd_tools/nc_to_mmd.py (hunk 1 above)
Comment (lhmarsden) [reply py_mmd_tools/nc_to_mmd.py]: Removed - sorry, I forgot to take this out.

**Comment by mortenwh** [comment]
Can you fix the flake errors and get the coverage back at 100% as well?

**Comment by mortenwh** [comment]
I'll make a new branch for this

**Pull request closed without merging**